l10n_it_intrastat_statement/models/intrastat_statement.py

```diff
@@ -819,7 +823,14 @@ def compute_statement(self):
                         *refund_section_details
                     )
                     to_refund_model = self.env[refund_section_model]
-                    self.refund_line(line, to_refund_model)
+                    if not (
+                        line.invoice_id.reversed_entry_id
+                        and (
+                            line.invoice_id.reversed_entry_id.date < period_date_start
+                            or line.invoice_id.reversed_entry_id.date > period_date_stop
+                        )
+                    ):
+                        self.refund_line(line, to_refund_model)
```

Comment:
Quindi la riga `line` viene segnata come rimborso solo quando la sua fattura ha un rimborso con data inclusa nel periodo, giusto?
Potresti modificarlo come suggerito qui sotto?

```suggestion
                    refund_date = line.invoice_id.reversed_entry_id.date
                    if refund_date and period_date_start <= refund_date <= period_date_stop:
                        self.refund_line(line, to_refund_model)
```
Credo sia un po' più leggibile.

Inoltre, non serve anche che il rimborso sia `intrastat`?
Forse si potrebbe cercare se il rimborso è tra le fatture delle righe che sono già state incluse nella dichiarazione qualche riga sopra, cosa ne dici? Così non serve rifare questi controlli sulle date.

Reply:
> Credo sia un po' più leggibile

Modifico

> Inoltre, non serve anche che il rimborso sia intrastat?

se `line.invoice_id` lo è, come può `line.invoice_id.reversed_entry_id` non esserlo?

Reply:
> > Inoltre, non serve anche che il rimborso sia intrastat?
> 
> se `line.invoice_id` lo è, come può `line.invoice_id.reversed_entry_id` non esserlo?

È un semplice flag sulla move, potrebbe esserci nella fattura ma non nel rimborso